### PR TITLE
Fixes typo, missing stars in #9469 and #9468

### PR DESCRIPTION
**Pull request by Jordach**
Fixes the typo listed in #9469 and #9468.

### CODE REVIEW

File: src/client/sky.cpp [hunk 1]
```diff
@@ -915,22 +915,7 @@ void Sky::setMoonTexture(std::string moon_texture,
...
-			);
-
-			star.normalize();
-			m_stars.emplace_back(star);
-		}
-	// Ignore changing star count if the new value is identical
-	} else if (m_star_params.count == star_count)
+	if (m_star_params.count == star_count && !force_update)
 		return;
 	else {
```
Comment (sfan5) [inline src/client/sky.cpp]: the else clause is still redundant

**Comment by sfan5** [comment]
Please also include the fix for #9475:
```diff
diff --git a/src/client/sky.cpp b/src/client/sky.cpp
index f2cab2f0..24d1dfaf 100644
--- a/src/client/sky.cpp
+++ b/src/client/sky.cpp
@@ -724,10 +724,10 @@ void Sky::draw_stars(video::IVideoDriver * driver, float wicked_time_of_day)
        if (m_star_params.starcolor.getAlpha() < 1)
                return;
 #if ENABLE_GLES
-       u16 *indices = new u16[m_star_count * 3];
+       u16 *indices = new u16[m_star_params.count * 3];
        video::S3DVertex *vertices =
-                       new video::S3DVertex[m_star_count * 3];
-       for (u32 i = 0; i < m_star_count; i++) {
+                       new video::S3DVertex[m_star_params.count * 3];
+       for (u32 i = 0; i < m_star_params.count; i++) {
                indices[i * 3 + 0] = i * 3 + 0;
                indices[i * 3 + 1] = i * 3 + 1;
                indices[i * 3 + 2] = i * 3 + 2;
@@ -750,8 +750,8 @@ void Sky::draw_stars(video::IVideoDriver * driver, float wicked_time_of_day)
                vertices[i * 3 + 2].Pos = p2;
                vertices[i * 3 + 2].Color = starcolor;
        }
-       driver->drawIndexedTriangleList(vertices.data(), m_star_count * 3,
-                       indices.data(), m_star_count);
+       driver->drawIndexedTriangleList(vertices, m_star_params.count * 3,
+                       indices, m_star_params.count);
        delete[] indices;
        delete[] vertices;
 #else
```

File: src/client/sky.cpp [hunk 2]
```diff
@@ -947,6 +930,7 @@ void Sky::setStarCount(u16 star_count, bool force_update)
 			m_stars.emplace_back(star);
 		}
 	}
+	return;
```
Comment (sfan5) [inline src/client/sky.cpp]: useless return statement

File: src/client/sky.cpp [hunk 3]
```diff
@@ -750,8 +750,8 @@ void Sky::draw_stars(video::IVideoDriver * driver, float wicked_time_of_day)
 		vertices[i * 3 + 2].Pos = p2;
 		vertices[i * 3 + 2].Color = starcolor;
 	}
-	driver->drawIndexedTriangleList(vertices.data(), m_star_count * 3,
-			indices.data(), m_star_count);
+	driver->drawIndexedTriangleList(vertices.data(), m_star_params.count * 3,
+			indices.data(), m_star_params.count);
```
Comment (sfan5) [inline src/client/sky.cpp]: You forgot to change `vertices.data()` and `indices.data()`

**Comment by paramat** [comment]
I confirm this fixes stars, did not test anything else though.

**Pull request merged**